ethereum/core/src/test-support/java/org/hyperledger/besu/ethereum/core/BlockDataGenerator.java

```diff
@@ -365,8 +371,22 @@ public UInt256 storageKey() {
     return receipts;
   }
 
+  public List<Log> logs(final int logsCount, final int topicsPerLog) {
+    return Stream.generate(() -> log(topicsPerLog)).limit(logsCount).collect(Collectors.toList());
+  }
+
   public Log log() {
-    return new Log(address(), bytesValue(5 + random.nextInt(10)), Collections.emptyList());
+    return log(0);
+  }
+
+  public Log log(final int topicCount) {
+    final List<LogTopic> topics =
+        Stream.generate(this::logTopic).limit(topicCount).collect(Collectors.toList());
+    return new Log(address(), bytesValue(5 + random.nextInt(10)), topics);
```

Comment:
Any idea on what these magic numbers mean? (I know you haven't added them, but if you know their meaning, it would be great to move them into constants with a meaningful name 😉 )

Reply:
@mbaxter wrote this but it looks like it's just to get random data in the log.

Reply:
Basically this is just generating random data between 5 and 15 bytes long.  We could add a method like `bytesValue(int minSize, int maxSize)` for clarity ...

Reply:
It already exists haha. Switched to that